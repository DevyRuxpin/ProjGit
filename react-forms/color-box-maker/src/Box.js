import React from 'react';

function Box({ id, width, height, backgroundColor, removeBox }) {
  const handleRemove = () => {
    removeBox(id);
  };

  const boxStyle = {
    width: `${width}px`,
    height: `${height}px`,
    backgroundColor,
    display: 'inline-block',
    margin: '10px',
    border: '1px solid black'
  };

  return (
    <div style={{ display: 'inline-block', margin: '10px' }}>
      <div style={boxStyle}></div>
      <button onClick={handleRemove}>X</button>
    </div>
  );
}

export default Box; 