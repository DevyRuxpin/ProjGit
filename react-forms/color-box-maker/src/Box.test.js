import { render, screen, fireEvent } from '@testing-library/react';
import Box from './Box';

const mockRemoveBox = jest.fn();

test('renders without crashing', () => {
  render(<Box id={1} width={100} height={100} backgroundColor="red" removeBox={mockRemoveBox} />);
});

test('matches snapshot', () => {
  const { asFragment } = render(<Box id={1} width={100} height={100} backgroundColor="red" removeBox={mockRemoveBox} />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders remove button', () => {
  render(<Box id={1} width={100} height={100} backgroundColor="red" removeBox={mockRemoveBox} />);
  expect(screen.getByText('X')).toBeInTheDocument();
});

test('calls removeBox when X button is clicked', () => {
  render(<Box id={1} width={100} height={100} backgroundColor="red" removeBox={mockRemoveBox} />);
  
  const removeButton = screen.getByText('X');
  fireEvent.click(removeButton);
  
  expect(mockRemoveBox).toHaveBeenCalledWith(1);
});

test('applies correct styles based on props', () => {
  render(<Box id={1} width={150} height={200} backgroundColor="blue" removeBox={mockRemoveBox} />);
  
  const boxDiv = screen.getByText('X').previousElementSibling;
  expect(boxDiv).toHaveStyle({
    width: '150px',
    height: '200px',
    backgroundColor: 'blue'
  });
}); 