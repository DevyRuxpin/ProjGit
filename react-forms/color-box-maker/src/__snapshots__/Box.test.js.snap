// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; margin: 10px;"
  >
    <div
      style="width: 100px; height: 100px; background-color: red; display: inline-block; margin: 10px; border: 1px solid black;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
